helix-core/src/main/java/org/apache/helix/controller/rebalancer/condition/RebalanceConditionBuilder.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.helix.controller.rebalancer.condition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+
+public class RebalanceConditionBuilder {
```

Comment:
Would you like to have different default condition impls?

For example, config change based condition, topology change based condition.

Reply:
Sure, can you share some code pointers on how to detect the config change or topology change?
e.g. for config change, is it something like the following:
```
event.getEventType() ==  ClusterEventType.ClusterConfigChange
```